extensions/vertx-redis/runtime/src/main/java/io/quarkus/vertx/redis/SyncRedisAPI.java

```diff
@@ -0,0 +1,417 @@
+package io.quarkus.vertx.redis;
+
+import java.util.List;
+
+import io.vertx.redis.client.Response;
+
+/**
+ * A synchronous RedisAPI offering blocking Redis commands.
+ * The commands have a default timeout of 10 seconds which can be configured
+ * via {@code quarkus.redis.timeout} configuration knob.
+ *
+ * For more information about how each individual command visit
+ * the <a href="https://redis.io/commands">Redis Commands Page</a>
+ */
+public interface SyncRedisAPI {
```

Comment:
I wonder if it makes sense to group these methods somehow. My reasoning is that we have a ton of methods so perhaps it would be useful to users to be able to more easily locate methods if they can be added to appropriate groups.

Reply:
This API design was heavily inspired by the mutiny based API. 

I'll be inclined to keep this dense to match the reactive counterpart. 

In the future this class be gone as IIRC there a plans in Vert.x to provide a blocking based API - I just cannot seem to find the issue/PR were I saw this plan.
/cc @pmlopes @cescoffier if you have any info.

Reply:
Hi, I don't know how you wrote this but on vertx upstream there's an handlebar template we have to generate this from the redis metadata. Perhaps you can reuse, improve it if you like

Reply:
> In the future this class be gone as IIRC there a plans in Vert.x to provide a blocking based API - I just cannot seem to find the issue/PR were I saw this plan.
> /cc @pmlopes @cescoffier if you have any info.

Found the issue - https://github.com/smallrye/smallrye-reactive-utils/issues/13 but it seem closed. 

> Hi, I don't know how you wrote this but on vertx upstream there's an handlebar template we have to generate this from the redis metadata. Perhaps you can reuse, improve it if you like

Well, IDE did most of the work for me really.